### DISCUSSION TITLE
Gnome keyring

### DISCUSSION THREAD

**Discussion by og900aero** [FreeRDP/FreeRDP#10830] [Q&A] [answered]
I use Debian 12 and freerdp (xreerdp2-x11 package version 2.10 under Debian 12). When I connect to the remote windows machine, ask password.I s there any way to save the password to the Gnome keyring? And the next time I reconnect, will it use the password saved in the gnome keyring?

**Comment by akallabeth** [comment]
@og900aero the `libsecret-tools` have a utility called `secret-tool` which allows to store/retrieve strings in gnome-keyring
you can wrap the `xfreerdp` call in a script to search for a `user@domain@host` or similar format and call `xfreerdp` with the extracted password.

you should use  the `/args-from: <file>|stdin|fd:<number>|env:<name>` argument in that script so the password does not show up in process listing.

**Reply by og900aero** [reply]
> @og900aero the `libsecret-tools` have a utility called `secret-tool` which allows to store/retrieve strings in gnome-keyring you can wrap the `xfreerdp` call in a script to search for a `user@domain@host` or similar format and call `xfreerdp` with the extracted password.
> 
> you should use the `/args-from: <file>|stdin|fd:<number>|env:<name>` argument in that script so the password does not show up in process listing.

Thx. I tried this, and working:
echo "$(secret-tool lookup rdp-server "ip" username "user")" | xfreerdp -grab-keyboard +clipboard -themes /u:user /v:ip:port /size:1910x1030 /window-position:1x35 /cert:ignore /microphone /from-stdin

But I dont understand /args-from switch. Maybe use it with this? /args-from:stdin ? Because I dont see any switch of name args-from in version 2.10.

**Reply by akallabeth** [reply, marked as answer]
@og900aero the `/args-from` is part of `FreeRDP 3.x` and is a (better) version of `/from-stdin`. you use it like `echo "/u:foo /p:pwd ..." | xfreerdp /args-from:stdin` (e.g. you can pipe all arguments in instead of providing them from command line)

**Reply by og900aero** [reply]
Ahaaa. Thx this information.
